src/mono/mono/mini/intrinsics.c

```diff
@@ -2322,6 +2322,63 @@ mini_emit_inst_for_method (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
 				return ins;
 			}
 		}
+	} else if ((cmethod->klass == mono_defaults.double_class) || (cmethod->klass == mono_defaults.single_class)) {
+		MonoGenericContext *method_context = mono_method_get_context (cmethod);
+		if (!strcmp (cmethod->name, "ConvertToIntegerNative") &&
+				method_context != NULL &&
+				method_context->method_inst->type_argc == 1) {
+			int opcode = 0;
+			MonoTypeEnum tto_type = method_context->method_inst->type_argv [0]->type;
+			MonoStackType tto_stack = STACK_I4;
+			switch (tto_type) {
+				case MONO_TYPE_I1:
+					opcode = OP_FCONV_TO_I1;
+					break;
+				case MONO_TYPE_I2:
+					opcode = OP_FCONV_TO_I2;
+					break;
+#if TARGET_SIZEOF_VOID_P == 4
```

Comment:
nit: in other places we use SIZEOF_REGISTER for these kinds of checks.

Reply:
looks like it varies based on a file, in `simd-intrinsics.c`, the `TARGET_SIZEOF_VOID_P` looks to be the dominant way.

Reply:
yeah its mixed, in this file SIZEOF_REGISTER seems to dominate. An interesting observation is that we have some code like this:

#if (SIZEOF_REGISTER > TARGET_SIZEOF_VOID_P) && (G_BYTE_ORDER == G_BIG_ENDIAN)

and we have this comment in [mono-machine.h](https://github.com/dotnet/runtime/blob/aeda1de6c20cb055e54c0527d595dfa919e2bfd6/src/mono/mono/utils/mono-machine.h#L64)

SIZEOF_REGISTER is usually the same as TARGET_SIZEOF_VOID_P, except when MONO_ARCH_ILP32 is defined